features/cellular/framework/device/CellularStateMachine.cpp

```diff
@@ -661,7 +661,7 @@ nsapi_error_t CellularStateMachine::start_dispatch()
 {
     MBED_ASSERT(!_queue_thread);
 
-    _queue_thread = new rtos::Thread(osPriorityNormal, 2048, NULL, "stm_queue");
+    _queue_thread = new rtos::Thread(osPriorityNormal, 2048 + 256, NULL, "stm_queue");
```

Comment:
This will just raise questions about these numbers. #define should be used here instead of magic number(s).

Reply:
@jarvte I don't know if you rather want to continue the discussion here or in the issue, I'll just post it here but if you rather want it in issue let me know!

Based on your previous comment, I did some more testing. The error I was originally getting was (see [log 141](https://github.com/ARMmbed/mbed-os/files/2899497/0141.log), line 390)

```
pre_event: wanted state is already / will be achieved, return without launching new event
```

Now, when I changed `_target_state` to `_state` that disappeared and I got a connection the 2nd time as well. Your previous comment got me thinking and I looked further in the disconnect.

- `_state_machine->reset` is called from `CellularDevice::cellular_callback` on a disconnect. It is actually called multiple times (see [log 140](https://github.com/ARMmbed/mbed-os/files/2899494/0140.log)) so I don't know it that is the expected behaviour.
- `CellularStateMachine::reset` doesn't reset the `_target_state`.
  When I add `_target_state = STATE_INIT;` here, I get a connection the 2nd time. Is the `_target_state` supposed to be reset too?

Reply:
@AnttiKauppila Makes sense, where do you propose I add this define, in the `CellularStateMachine` header file or is there a generic config file where this define should be added?
I propose making it something like `#define STM_QUEUE_THREAD_STACK_SIZE (2048 + 256)`. What do you think?

Reply:
@AnttiKauppila See a28bcae, let me know what you think!

Reply:
@vidavidorra That is fine, maybe those numbers could be counted together as that is the total stack size needed.

Reply:
2048 should be enough. It's not enough if application reserves memory in callbacks or have lot's of traces in callbacks. I updated doxygen about this in pr(https://github.com/ARMmbed/mbed-os/pull/9898) which fixes the connect-disconnect issues.